src/+otp/+constrainedpendulum/+presets/Canonical.m

```diff
@@ -0,0 +1,22 @@
+classdef Canonical < otp.constrainedpendulum.ConstrainedPendulumProblem
+    %CANONICAL The constrained pendulum problem
+    %
+    % See
+    %    Ascher, Uri. "On symmetric schemes and differential-algebraic equations."
+    %    SIAM journal on scientific and statistical computing 10.5 (1989): 937-949.
+    
+    methods
+        function obj = Canonical
+            tspan = [0; 10];
+            
+            params = otp.constrainedpendulum.ConstrainedPendulumParameters;
+            params.Mass = 1;
+            params.Length = 1;
+            params.Gravity = otp.utils.PhysicalConstants.EarthGravity;
+            
+            y0 = [sqrt(2)/2; sqrt(2)/2; 0; 0];
+            
+            obj = obj@otp.constrainedpendulum.ConstrainedPendulumProblem(tspan, y0, params);
```

Comment:
Will the `solve` functionality be disabled for this problem?

Reply:
That is an interesting question. In theory we should be able to add our own small method to OTP to handle these cases

Reply:
Good point.  Here's one idea: Add a `HighIndexDAE` property to https://github.com/ComputationalScienceLaboratory/ODE-Test-Problems/blob/master/src/%2Botp/%2Butils/Solver.m which simply throws an error message that high-index DAEs are not supported by matlab solvers.  Eventually when we have a constrained mechanical system problem, we will support RHS reduced to index-1 which can rely on built-in solvers.